source/plugins/go/input/containerinventory/containerinventory.go

```diff
@@ -0,0 +1,183 @@
+}
+
+type containerInventoryPlugin struct {
+	tag         string
+	runInterval int
+}
+
+var (
+	FLBLogger                 *log.Logger
+	namespaceFilteringMode    = "off"
+	namespaces                []string
+	addonTokenAdapterImageTag = ""
+	agentConfigRefreshTracker = time.Now().Unix()
+	tag                       = "oneagent.containerInsights.CONTAINER_INVENTORY_BLOB"
+	runInterval               = 60
+	containerType             = os.Getenv("CONTAINER_TYPE")
+	hostName                  = ""
+	telemetryTimeTracker      int64
+	isFromCache	       = false
+)
+
+// Init An instance of the configuration loader will be passed to the Init method so all the required
+// configuration entries can be retrieved within the plugin context.
+func (p *containerInventoryPlugin) Init(ctx context.Context, fbit *plugin.Fluentbit) error {
+	if fbit.Conf.String("tag") == "" {
+		p.tag = tag
+	} else {
+		p.tag = fbit.Conf.String("tag")
+	}
+	if fbit.Conf.String("run_interval") == "" {
+		p.runInterval = runInterval
+	} else {
+		p.runInterval, _ = strconv.Atoi(fbit.Conf.String("run_interval"))
+	}
+
+	osType := os.Getenv("OS_TYPE")
+	if strings.EqualFold(osType, "windows") {
+		FLBLogger = lib.CreateLogger("/etc/amalogswindows/fluent-bit-input.log")
+	} else {
+		FLBLogger = lib.CreateLogger("/var/opt/microsoft/docker-cimprov/log/fluent-bit-input.log")
+	}
+
+	return nil
+}
+
+// Collect this method will be invoked by the fluent-bit engine after the initialisation is successful
+// this method can lock as each plugin its implemented in its own thread. Be aware that the main context
+// can be cancelled at any given time, so handle the context properly within this method.
+// The *ch* channel parameter, is a channel handled by the runtime that will receive messages from the plugin
+// collection, make sure to validate channel closure and to follow the `plugin.Message` struct for messages
+// generated by plugins.
+func (p containerInventoryPlugin) Collect(ctx context.Context, ch chan<- plugin.Message) error {
+	tick := time.NewTicker(time.Duration(p.runInterval) * time.Second)
+
+	for {
+		select {
+		case <-ctx.Done():
+			err := ctx.Err()
+			if err != nil && !errors.Is(err, context.Canceled) {
+				return err
+			}
+
+			return nil
+		case <-tick.C:
+			emitTime := time.Now()
+			telemetryTimeTracker = emitTime.Unix()
+			FLBLogger.Print("containerinventory::enumerate.start @ ", time.Now().UTC().Format(time.RFC3339))
+			messages := p.enumerate()
+			FLBLogger.Print("containerinventory::enumerate.end @ ", time.Now().UTC().Format(time.RFC3339))
+
+			ch <- plugin.Message{
+				Record: map[string]any {
+					"tag":     tag,
+					"messages": messages,
+				},
+				Time: emitTime,
+			}
+			FLBLogger.Print("containerinventory::emitted ", len(messages) ," container inventory records @ ", time.Now().UTC().Format(time.RFC3339))
+
+			timeDifference := int(math.Abs(float64(time.Now().Unix() - telemetryTimeTracker)))
```

Comment:
I see telemetryTimeTracker being reset (every interval, 60s) at line 86 - https://github.com/microsoft/Docker-Provider/blob/96f3dc0d84b20f90b86ce61e6b3b5b4e9bf72441/source/plugins/go/input/containerinventory/containerinventory.go#L86  which means timedifference always 1 min. Please check this logic.